src/main/java/com/redhat/digkins/DiggerClient.java

```diff
@@ -0,0 +1,56 @@
+package com.redhat.digkins;
+
+import com.offbytwo.jenkins.JenkinsServer;
+import com.redhat.digkins.services.CreateJobService;
+import com.redhat.digkins.util.JenkinsAuth;
+import com.redhat.digkins.util.DiggerClientException;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+/**
+ * Digger Java Client
+ * <p>
+ * Interact with digger jenkins api!
+ */
+public class DiggerClient {
+
+  private final JenkinsServer jenkins;
+
+  public DiggerClient(JenkinsAuth auth) throws URISyntaxException {
+    this.jenkins = new JenkinsServer(new URI(auth.getUrl()), auth.getUser(), auth.getPassword());
+  }
+
+  /**
+   * Create new digger job on jenkins platform
+   *
+   * @param name      - job name that can be used later to reference job
+   * @param gitRepo   - git repository url (full git repository url. e.g git@github.com:wtrocki/helloworld-android-gradle.git
+   * @param gitBranch - git repository branch (default branch used to checkout source code)
+   */
+  public void createJob(String name, String gitRepo, String gitBranch) throws DiggerClientException {
+    CreateJobService service = new CreateJobService(this.jenkins);
```

Comment:
Good point of view. I'm using here short lived objects because of the way we going to use this library. When creating client we would use it to do only one job (just calling one Service).Having all services created upfront as part of DiggerClient would mean that we would require x objects (where x is number of services) where actually only one Service would be required. Happy to change this approach - no strong feelings here.